src/hdlc.rs

```diff
@@ -0,0 +1,62 @@
+use alloc::borrow::{Cow, ToOwned};
+use hdlcparse::{
+  llc::{parse_llc, Llc},
+  type3::HdlcFrame,
+};
+
+use crate::{DlmsDataLinkLayer, Error};
+
+#[derive(Debug, Default)]
+pub struct HdlcDataLinkLayer {
+  is_server: bool,
+}
+
+impl HdlcDataLinkLayer {
+  pub fn new(is_server: bool) -> Self {
+    Self { is_server }
+  }
+  fn validate_llc(&self, llc: Llc) -> bool {
+    llc.dest_sap == 0xE6
+      && (llc.src_sap == if self.is_server { 0xE6 } else { 0xE7 })
```

Comment:
Is the `is_server` variable really needed here? `dest_sap` has to be `0xE6`, so can `src_sap` even be anything other than `0xE7`?

Reply:
Yes if it is a server (smart meter). So a "normal" client would expect `E6E7E00` and a server `E6E600`